### PR TITLE
Refactor/error management

### PR DESCRIPTION
**Pull request by Eikix**
Add error management to the test and list commands

### CODE REVIEW

File: src/cli/commands/list/tests.rs [hunk 1]
```diff
@@ -27,10 +27,6 @@ fn returns_error_in_case_of_failure() {
 	.exec();
 
 	assert!(result.is_err());
```
Comment (tdelabro) [inline src/cli/commands/list/tests.rs:29]: Je te conseille l'excellente lib https://doc.rust-lang.org/std/assert_matches/macro.assert_matches.html
pour assert que des struct matchent bien d'autres. 
Donc la notament tu pourras match le type de ton erreur ainsi que son variant

File: src/cli/commands/mod.rs [hunk 2]
```diff
@@ -51,11 +60,17 @@ impl fmt::Display for Output {
...
-impl CommandExecution<Output> for Commands {
-	fn exec(&self) -> Result<Output, String> {
+impl CommandExecution<Output, CommandError> for Commands {
+	fn exec(&self) -> Result<Output, CommandError> {
 		match &self {
-			Commands::List(args) => args.exec().map(|o| Output(CommandOutputs::List(o))),
-			Commands::Test(args) => args.exec().map(|o| Output(CommandOutputs::Test(o))),
+			Commands::List(args) => args
+				.exec()
+				.map_err(CommandError::ListCommandError)
```
[the remark this answers is not in the document]
Comment (tdelabro) [reply src/cli/commands/mod.rs:68]: Et tu as aussi ce crate: https://docs.rs/mapinto/latest/mapinto/trait.ResultMapErrInto.html
Qui fourni une fonction qui fait le café tout seul

File: src/cli/commands/test/mod.rs [hunk 3]
```diff
@@ -20,21 +20,43 @@ use colored::Colorize;
...
 
+#[derive(Error, Debug)]
+pub enum TestCommandError {
+	#[error("Failed to list test entrypoints for file {0}: {1}")]
+	ListEntrypointsError(PathBuf, String),
+	#[error("Failed to compile file {0}: {1}")]
+	RunTestError(String, PathBuf, String),
+	#[error(transparent)]
+	IOError(#[from] io::Error),
+	// transparent error for json serde
```
Comment (tdelabro) [inline src/cli/commands/test/mod.rs:47]: Why a comment just for this one ?

File: src/cli/commands/test/mod.rs [hunk 4]
```diff
@@ -20,21 +20,43 @@ use colored::Colorize;
...
 use super::{
-	list::{path_is_valid_directory, ListArgs, ListOutput},
+	list::{path_is_valid_directory, ListArgs, ListCommandError, ListOutput},
 	CommandExecution,
 };
 
 use crate::{
 	cairo_run::cairo_run,
-	compile::compile,
+	compile::{compile, Error},
```
Comment (tdelabro) [inline src/cli/commands/test/mod.rs:34]: alias le avec `compile::{compile, Error as CompileError}`
Trop de chose ont tendance à s'appeler `Error`

Comment (tdelabro) [reply src/cli/commands/test/mod.rs:34]: ou alors ne l'importe pas et garde le node précedent du path `compile::Error`

L'un ou l'aute sont OK

File: src/hints/expect_revert/tests.rs [hunk 5]
```diff
@@ -1,11 +1,15 @@
...
-fn test_expect_revert(#[case] path: &str, #[case] expected_success: bool) {
+fn test_expect_revert(
+	#[case] path: &str,
+	#[case] expected_success: bool,
+) -> Result<(), TestCommandError> {
 	let path = std::path::PathBuf::from(path);
-	let (_, actual_success) = run_single_test("test_expect_revert", &path);
+	let (_, actual_success) = run_single_test("test_expect_revert", &path)?;
 	assert_eq!(expected_success, actual_success);
+	Ok(())
```
Comment (tdelabro) [inline src/hints/expect_revert/tests.rs:14]: Why do you need to return an error?
In test we usually just panic when something does not go according to the plan. And it tag the test as FAILED.

`let (_, actual_success) = run_single_test("test_expect_revert", &path).expect("`run_single_test` should be Ok");`

**Review by tdelabro — APPROVED** [review]

File: src/cli/commands/list/tests.rs (hunk 1 above)
Comment (Eikix) [reply src/cli/commands/list/tests.rs:29]: This is a nightly build, wdyt about implementing unstable libs?

File: src/cli/commands/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Eikix) [reply src/cli/commands/mod.rs:68]: Dacc!

File: src/cli/commands/list/tests.rs (hunk 1 above)
Comment (tdelabro) [reply src/cli/commands/list/tests.rs:29]: my bad. https://docs.rs/assert_matches/1.5.0/assert_matches/
This is the crate. It run in stable channel

Comment (tdelabro) [reply src/cli/commands/list/tests.rs:29]: I see no problem using the nightly channel. 
Stable features won't be affected by it. They will stay the same. 
But you will gain access to those nightly features. Those are subject to breaking changes and can even be abandoned. You won't have a guarantee they will ever end up in the stable release. And it's sometimes the only place you can get the feature you need. 
So if you need it, use it. No problem with it.

File: src/cli/commands/list/tests.rs [hunk 6]
```diff
@@ -1,5 +1,7 @@
+use assert_matches::assert_matches;
```
Comment (tdelabro) [inline src/cli/commands/list/tests.rs:1]: You imported it but did not used it|

`assert!(result.is_err());`

**Review by tdelabro — APPROVED** [review]

**Pull request merged**